src/rust/catpowder/win/runtime.rs

```diff
@@ -57,16 +62,26 @@ impl SharedCatpowderRuntime {
     /// Instantiates a new XDP runtime.
     pub fn new(config: &Config) -> Result<Self, Fail> {
         let ifindex: u32 = config.local_interface_index()?;
-        const QUEUEID: u32 = 0; // We do no use RSS, thus queue id is always 0.
 
         trace!("Creating XDP runtime.");
         let mut api: XdpApi = XdpApi::new()?;
 
         // Open TX and RX rings
-        let tx: TxRing = TxRing::new(&mut api, Self::RING_LENGTH, ifindex, QUEUEID)?;
-        let rx: RxRing = RxRing::new(&mut api, Self::RING_LENGTH, ifindex, QUEUEID)?;
+        let tx: TxRing = TxRing::new(&mut api, Self::RING_LENGTH, ifindex, 0)?;
+
+        let sys_proc_count: usize = count_processor_cores()?;
+        let mut rx_rings: Vec<RxRing> = Vec::with_capacity(sys_proc_count);
+        rx_rings.push(RxRing::new(&mut api, Self::RING_LENGTH, ifindex, 0)?);
```

Comment:
Can this be moved inside the loop? Also, if this fails, the function would fail, but inside the loop, failures are ignored when creating RxRings for other queueids. Is it ok to let those errors be ignored?

Reply:
This is the caveat mentioned in the PR description. The only trivial way we have to determine the number of RSS queues is to try to bind to each subsequent index until bind fails.
There are three ways we could have done this:
1. As is; treat all failures on queue binding >0 as an indication that we've reached the max RSS queue ID.
2. Same as 1, but refactor `RxQueue` and `Socket` to inform on error specifically from `XskBind`. We still have no way to determine failures from an invalid RSS queue ID versus other failures with `XskBind`, so this still can have mis-identified errors.
3. We can add infrastructure to query MSI to determine the RSS settings. This is probably a better solution, but is quite complex and will take some doing to get right.

I chose 1 for expediency, with the expectation of targeting a solution like 3 in the future.

Reply:
My preference would be to leave this as-is for now.